app/lib/tariff_synchronizer/base_update.rb

```diff
@@ -140,28 +150,44 @@ class << self
       delegate :instrument, to: ActiveSupport::Notifications
 
       def sync
-        (pending_from..Date.current).each { |date| download(date) }
+        applicable_download_date_range.each { |date| download(date) }
+
         notify_about_missing_updates if last_updates_are_missing?
       end
 
       def update_type
         raise 'Update Type should be specified in inheriting class'
       end
 
-    private
+      def applicable_download_date_range
+        download_start_date..download_end_date
+      end
 
-      def pending_from
-        if last_download = (last_pending || descending.first)
-          last_download.issue_date
-        else
+      private
+
+      def download_end_date
+        Time.zone.today
+      end
+
+      def download_start_date
+        if pending_applied_and_failed.count.zero?
           TariffSynchronizer.initial_update_date_for(update_type)
+        else
+          last_download = (last_pending || last_applied || last_failed)
+
+          # Handle missing downloads when older downloads came late
+          if last_download.issue_date < DOWNLOAD_FROM.ago.to_date
+            last_download.issue_date
+          else
+            DOWNLOAD_FROM.ago.to_date
+          end
```

Comment:
is not this the same as:
`[last_download.issue_date, DOWNLOAD_FROM.ago.to_date].min`
?

Reply:
Yeah, it is the same. Not sure which is clearer. I guess yours is terser.